### DISCUSSION TITLE
Hyperparameter tuning

### DISCUSSION THREAD

**Discussion by mango-picket** [aws/sagemaker-python-sdk#3863] [Help]
Hi Everyone,

Context on the problem:
I am training an SK Learn pipeline model with custom pre-processing and an XGBRegressor model under an SKLearn Estimator image in a custom bring my own training script framework. 
I have a training and validation split of my data that I am able to load in my training script, train the pipeline model, make predictions on training and validation sets, and also able to log the evaluation metrics like train:mae, val:mae etc...
I am also able to launch a hyper-parameter tuning job with my custom metric-definitions and the job finishes successfully. The objective I choose to minimize is validaiton:mae with "Bayesian" optimization strategy.

Questions:
1. Why does each training job associated with the tuning job have a 'minimum', 'maximum', 'average', 'std-dev' of the metrics? I do not perform any cross-validation in my custom training script, so where does sagemaker get these numbers?
2. Why does the validation error that the metrics table log the minimum value? and not the average value? How does this not lead to overfitting if we drive the tuning objective to minimum while selecting the minimum error split from internal cross-validation.

Please help me understand what's going on in sagemaker hyper-parameter tuning metric logs and objective tuning. I understand bayesian optimization and sagemaker seems to be doing the optimization correctly, but it seems that sagemaker is not using the correct numbers to optimize.
